tests/Client/UpdaterTest.php

```diff
@@ -288,6 +287,32 @@ public function testVerifiedDelegatedDownload(): void
         }
     }
 
+    /**
+     * Tests for enforcement of maximum number of roles limit.
+     */
+    public function testMaximumRoles(): void
+    {
+        $fixturesSet = 'TUFTestFixtureNestedDelegated';
+        $fileName = 'level_1_2_terminating_3_target.txt';
+
+        // Ensure the file can found if the maximum role limit is 100.
+        $this->localRepo = $this->memoryStorageFromFixture($fixturesSet, 'tufclient/tufrepo/metadata/current');
+        $this->testRepo = new TestRepo($fixturesSet);
+        $updater = $this->getSystemInTest($fixturesSet);
+        $testFilePath = static::getFixturesRealPath($fixturesSet, "tufrepo/targets/$fileName", false);
+        $testFileContents = file_get_contents($testFilePath);
+        self::assertNotEmpty($testFileContents);
+        $this->assertSame($testFileContents, $updater->download($fileName)->wait()->getContents());
```

Comment:
Not blocking, but: why do we use `self::assertStuff()` and `$this->assertStuff()` interchangeably? Should we just choose one and stick with it?

Reply:
no reason just  copied different line from different places. will make a PR to clean this up

Reply:
fixed though here